packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/executors/local/executor.py

```
@@ -627,6 +622,190 @@ def kill_job(job_id: str) -> None:
+                        except (OSError, IOError) as e:
+                            # File might have been deleted or moved
+                            # Don't log error for every check, only on first error
+                            if log_info.get("error_printed", False) is False:
+                                logger.error(f"Error reading {log_info['path']}: {e}")
+                                log_info["error_printed"] = True
+                            log_info["file_handle"] = None
+                        except Exception:
+                            # Reset error flag if we successfully read again
+                            log_info["error_printed"] = False
+
+                # If no activity, sleep briefly to avoid busy waiting
+                if not any_activity:
+                    time.sleep(0.1)
+
+        except KeyboardInterrupt:
+            # Clean exit on Ctrl+C
+            pass
+        finally:
+            # Close all file handles
+            for log_info in log_files:
+                if log_info["file_handle"] is not None:
+                    try:
+                        log_info["file_handle"].close()
+                    except Exception:
+                        pass
+
+    @staticmethod
+    def _extract_task_name(job_data: JobData, job_id: str) -> str:
+        """Extract task name from job data config.
+
+        Args:
+            job_data: JobData object containing config.
+            job_id: Job ID for error reporting.
+
+        Returns:
+            Task name string.
+        """
+        config = job_data.config or {}
+        evaluation = config.get("evaluation", {}) or {}
+        tasks = evaluation.get("tasks", []) or []
+
+        # Find the task that matches this job
+        # For job_id like "15b9f667.0", index is 0
+        try:
+            if "." in job_id:
+                index = int(job_id.split(".")[1])
+                if 0 <= index < len(tasks):
```

Should we raise an error here?